contracts/src/systems/combat/contracts.cairo

```diff
@@ -620,7 +621,7 @@ mod combat_systems {
             let defending_army_position: Position = get!(world, defending_army_id, Position);
             attacking_army_position.assert_same_location(defending_army_position.into());
 
-            let battle_id: u128 = world.uuid().into();
+            let battle_id: ID = world.uuid().into();
```

Comment:
no need to use `into()` I think, `uuid()` returns a `usize` which I believe is a `u32`